src/network.rs

```diff
@@ -413,7 +412,21 @@ impl EventLoop {
 							.add_address(&peer_id, multiaddr);
 					}
 				},
-				_ => (),
+				MdnsEvent::Expired(addrs_list) => {
+					for (peer_id, multiaddr) in addrs_list {
+						debug!(
+							"MDNS got expired peer with ID: {:#?} and Address: {:#?}",
+							peer_id, multiaddr
+						);
+
+						if !self.swarm.behaviour_mut().mdns.has_node(&peer_id) {
```

Comment:
Should we also check whether we have active open connections?

Reply:
because of the keep_alive behavior, the MDNS record will only expire on closed connections
this line also checks if the record is still active before remove